### PR TITLE
New vector using react-sdk

### PR DESCRIPTION
**Pull request by dbkr**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,145 +10,33 @@ Getting started
...
-5. Switch to the example directory: `cd examples/vector`
-6. Install the example app prerequisites: `npm install`
-7. Build the example and start a server: `npm start`
+5. Start the development builder and a testing server: `npm start`
+6. Wait a few seconds for the initial build to finish.
+7. Open http://127.0.0.1:8080/ in your browser to see your newly built Vector.
 
-Now open http://127.0.0.1:8080/ in your browser to see your newly built
-Vector.
+Wiht `npm start`, Any changes you make to the source files will cause a rebuild so
```
Comment (ara4n) [inline README.md]: s/Wiht/with/, s/Any/any/

File: src/skins/vector/skindex.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+ * THIS FILE IS AUTO-GENERATED
+ * You can edit it you like, but your changes will be overwritten,
+ * so you'd just be trying to swim upstream like a salmon.
+ * You are not a salmon.
```
Comment (ara4n) [inline src/skins/vector/skindex.js]: :D

File: src/skins/vector/views/molecules/EventAsTextTile.js [hunk 3]
```diff
@@ -18,21 +18,23 @@ limitations under the License.
...
 
     render: function() {
+        var MessageTimestamp = sdk.getComponent('atoms.MessageTimestamp');
+        var MemberAvatar = sdk.getComponent('atoms.MemberAvatar');
+
         var text = TextForEvent.textForEvent(this.props.mxEvent);
         if (text == null || text.length == 0) return null;
 
+        var MessageTimestamp = sdk.getComponent('atoms.MessageTimestamp');
+        var MemberAvatar = sdk.getComponent('atoms.MemberAvatar');
```
Comment (ara4n) [inline src/skins/vector/views/molecules/EventAsTextTile.js]: Wut? duplication og lines 30 & 31 shirley?

File: src/skins/vector/views/molecules/RoomCreate.js [hunk 4]
```diff
@@ -17,11 +17,8 @@ limitations under the License.
 'use strict';
 
 var React = require('react');
-var classNames = require('classnames');
 
-//var RoomCreateController = require("../../../../src/controllers/molecules/RoomCreateController");
-
-var MatrixClientPeg = require("../../../../src/MatrixClientPeg");
+//var RoomCreateController = require('matrix-react-sdk/lib/controllers/molecules/RoomCreateController')
```
Comment (ara4n) [inline src/skins/vector/views/molecules/RoomCreate.js]: what's with these commented out controllers?

**Comment by ara4n** [comment]
lgtm beyond comments above. i haven't tried to review the split between vector-web & matrix-react-sdk here.

File: src/skins/vector/views/molecules/EventAsTextTile.js (hunk 3 above)
Comment (dbkr) [inline src/skins/vector/views/molecules/EventAsTextTile.js]: Oops, thanks.

File: src/skins/vector/views/molecules/RoomCreate.js (hunk 4 above)
Comment (dbkr) [inline src/skins/vector/views/molecules/RoomCreate.js]: I think this was from people noting that the controllers were empty and so could just be done away with altogether. I guess they may as well go.

File: package.json [hunk 5]
```diff
@@ -1,40 +1,47 @@
...
-    "prepublish": "npm run build"
+    "reskindex": "reskindex vector -h src/skins/vector/header",
+    "build:css": "catw 'src/skins/vector/css/**/*.css' -o vector/bundle.css -c uglifycss --no-watch",
+    "build:compile": "babel --source-maps -d lib src",
+    "build:bundle": "NODE_ENV=production webpack -p lib/vector/index.js vector/bundle.js",
+    "build": "npm run build:css && npm run build:compile && npm run build:bundle",
+    "start:js": "webpack -w src/vector/index.js vector/bundle.js",
+    "start:skins:css": "catw 'src/skins/vector/css/**/*.css' -o vector/bundle.css",
+    "//cache": "Note the -c 1 below due to https://code.google.com/p/chromium/issues/detail?id=508270",
+    "start": "parallelshell 'npm run start:js' 'npm run start:skins:css' 'http-server -c 1 vector'",
```
Comment (Kegsay) [inline package.json]: Single-quotes like this will prevent this from working on Windows. It needs to be `\"`.

File: config.json [hunk 6]
```diff
@@ -0,0 +1,4 @@
+{
+    "default_hs_url": "https://matrix.org",
+    "default_is_url": "https://vector.im"
```
Comment (Kegsay) [inline config.json]: Where are these config values used? I don't see usages anywhere in this PR.

File: package.json (hunk 5 above)
Comment (dbkr) [inline package.json]: Ah, thanks

File: config.json (hunk 6 above)
Comment (dbkr) [inline config.json]: It's in there, in Login.js & Register.js

**Pull request merged**